rosdep install misses some dependencies?

after cloning and rosdep install'ing, it appears that there are still some missing dependencies? 

--
$ catkin_make

In file included from /home/dereck/Dropbox/ros/hydro/catkin/src/OctoSlam/src/OctoSlamCalcs.cpp:1:0:
/home/dereck/Dropbox/ros/hydro/catkin/src/OctoSlam/src/OctoSlamCalcs.h:4:23: fatal error: Eigen/Dense: No such file or directory
compilation terminated.
In file included from /home/dereck/Dropbox/ros/hydro/catkin/src/OctoSlam/src/transformScan.cpp:2:0:
/home/dereck/Dropbox/ros/hydro/catkin/src/OctoSlam/src/OctoSlamCalcs.h:4:23: fatal error: Eigen/Dense: No such file or directory
compilation terminated.
In file included from /home/dereck/Dropbox/ros/hydro/catkin/src/OctoSlam/src/roundVoxel.cpp:1:0:
/home/dereck/Dropbox/ros/hydro/catkin/src/OctoSlam/src/OctoSlamCalcs.h:4:23: fatal error: Eigen/Dense: No such file or directory
compilation terminated.
make[2]: *** [OctoSlam/CMakeFiles/OctoSlam.dir/src/transformScan.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [OctoSlam/CMakeFiles/OctoSlam.dir/src/OctoSlamCalcs.cpp.o] Error 1
make[2]: *** [OctoSlam/CMakeFiles/OctoSlam.dir/src/roundVoxel.cpp.o] Error 1
In file included from /home/dereck/Dropbox/ros/hydro/catkin/src/OctoSlam/src/Localization.cpp:2:0:
/home/dereck/Dropbox/ros/hydro/catkin/src/OctoSlam/src/OctoSlamCalcs.h:4:23: fatal error: Eigen/Dense: No such file or directory

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,26 +8,22 @@ find_package(catkin REQUIRED COMPONENTS
   roscpp
   rospy
   std_msgs
+  tf
+  laser_geometry
 )
 
 find_package(Eigen REQUIRED)
 include_directories(${Eigen_INCLUDE_DIRS})
 
-
 find_package(octomap REQUIRED)
 include_directories(${OCTOMAP_INCLUDE_DIRS})
 link_libraries(${OCTOMAP_LIBRARIES})
 link_directories(${OCTOMAP_LIBRARY_DIRS})
+
 include_directories(
   ${catkin_INCLUDE_DIRS}
 )
-add_executable(OctoSlam src/OctoSlam.cpp src/OctoSlamCalcs.cpp src/Localization.cpp
-        src/roundVoxel.cpp src/transformScan.cpp) 
-#add_executable(Tester src/Tester.cpp Scan3.cpp src/transformScan.cpp)
-#add_executable(read_map src/practice/read_map.cpp)
-target_link_libraries(OctoSlam
-   ${catkin_LIBRARIES}
-)
+
 #target_link_libraries(read_map ${OCTOMAP_LIBRARIES})
 # System dependencies are found with CMake's conventions
 # find_package(Boost REQUIRED COMPONENTS system)
@@ -118,8 +114,18 @@ catkin_package(
 # )
 
 ## Declare a cpp executable
-# add_executable(octoslam_node src/octoslam_node.cpp)
-#add_executable(Scan3 src/Scan3.cpp)
+add_executable(OctoSlam src/OctoSlam.cpp src/OctoSlamCalcs.cpp src/Localization.cpp
+        src/roundVoxel.cpp src/transformScan.cpp) 
+       
+target_link_libraries(OctoSlam
+   ${catkin_LIBRARIES}
+)
+
+add_executable(scan_to_cloud src/scan_to_cloud.cpp) 
+target_link_libraries(scan_to_cloud
+   ${catkin_LIBRARIES}
+)
+
 
 ## Add cmake target dependencies of the executable/library
 ## as an example, message headers may need to be generated before nodes
@@ -142,11 +148,11 @@ catkin_package(
 # )
 
 ## Mark executables and/or libraries for installation
-# install(TARGETS octoslam octoslam_node
+ install(TARGETS OctoSlam scan_to_cloud
 #   ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
 #   LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-#   RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-# )
+   RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+ )
 
 ## Mark cpp header files for installation
 # install(DIRECTORY include/${PROJECT_NAME}/
@@ -156,11 +162,11 @@ catkin_package(
 # )
 
 ## Mark other files for installation (e.g. launch and bag files, etc.)
-# install(FILES
-#   # myfile1
+ install(FILES
+    build_map.launch
 #   # myfile2
-#   DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
-# )
+   DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+ )
 
 #############
 ## Testing ##

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,11 @@ find_package(catkin REQUIRED COMPONENTS
   rospy
   std_msgs
 )
+
+find_package(Eigen REQUIRED)
+include_directories(${Eigen_INCLUDE_DIRS})
+
+
 find_package(octomap REQUIRED)
 include_directories(${OCTOMAP_INCLUDE_DIRS})
 link_libraries(${OCTOMAP_LIBRARIES})

--- a/build_map.launch
+++ b/build_map.launch
@@ -1,12 +1,12 @@
 <launch>
   <!-- convert the scan data into point clouds --> 
-  <node pkg="mapping_3D" type="scan_to_cloud" name="scan_to_cloud" />
+  <node pkg="octoslam" type="scan_to_cloud" name="scan_to_cloud" />
 
   <!-- run the octomap node to build map-->
   <node pkg="octomap_server" type="octomap_server_node" name="octomap_server">
     <param name="resolution" value=".025" />
 
-    <param name="frame_id" type="string" value="/vrpn" />
+    <param name="frame_id" type="string" value="/vrpn" /> 
 
     <param name="max_sensor_range" value="4.0" />
 

--- a/package.xml
+++ b/package.xml
@@ -45,6 +45,7 @@
   <build_depend>std_msgs</build_depend>
   <build_depend>octomap</build_depend>
   <build_depend>eigen</build_depend>
+  <build_depend>laser_geometry</build_depend>
 
   
   <run_depend>roscpp</run_depend>
@@ -52,6 +53,7 @@
   <run_depend>std_msgs</run_depend>
   <run_depend>octomap</run_depend>
   <run_depend>eigen</run_depend>
+  <run_depend>laser_geometry</run_depend>
 
 
   <!-- The export tag contains other, unspecified, tags -->

--- a/package.xml
+++ b/package.xml
@@ -44,10 +44,14 @@
   <build_depend>rospy</build_depend>
   <build_depend>std_msgs</build_depend>
   <build_depend>octomap</build_depend>
+  <build_depend>eigen</build_depend>
+
+  
   <run_depend>roscpp</run_depend>
   <run_depend>rospy</run_depend>
   <run_depend>std_msgs</run_depend>
   <run_depend>octomap</run_depend>
+  <run_depend>eigen</run_depend>
 
 
   <!-- The export tag contains other, unspecified, tags -->

